common/lib/xmodule/xmodule/modulestore/split_mongo/split_draft.py

```diff
@@ -198,6 +198,13 @@ def delete_item(self, location, user_id, revision=None, **kwargs):
                 branches_to_delete = [ModuleStoreEnum.BranchName.published, ModuleStoreEnum.BranchName.draft]
             elif revision is None:
                 branches_to_delete = [ModuleStoreEnum.BranchName.draft]
+                delete_children_catogries = ['chapter', 'sequential', 'vertical']
```

Comment:
I don't understand. What do you mean "those categories which contain only xblock children"? Do you mean these are the categories that could contain children? If that's what you mean, then it's not right (for example, what about `course`?). Also, if that's what you're trying to do, then you should ask XBlock to load the class for `location.category`, and then check the class for the `has_children` attribute, so that future XBlock types that can have children are also caught by this logic.

It should almost never be the case that platform code should be checking against specific hard-coded XBlock types. The platform should check against properties of the XBlock, so that future XBlocks that meet the same criteria are also considered.